libyul/AST.h

```diff
@@ -38,20 +38,34 @@ using Type = YulString;
 
 struct DebugData
 {
-	explicit DebugData(langutil::SourceLocation _location, std::optional<int64_t> _astID = {}):
-		location(std::move(_location)),
+	explicit DebugData(
+		langutil::SourceLocation _irLocation,
+		langutil::SourceLocation _originalLocation = {},
+		std::optional<int64_t> _astID = {}
+	):
+		irLocation(std::move(_irLocation)),
+		originalLocation(std::move(_originalLocation)),
 		astID(std::move(_astID))
 	{}
 
 	static std::shared_ptr<DebugData const> create(
-		langutil::SourceLocation _location = {},
+		langutil::SourceLocation _irLocation = {},
+		langutil::SourceLocation _originalLocation = {},
 		std::optional<int64_t> _astID = {}
 	)
 	{
-		return std::make_shared<DebugData const>(std::move(_location), std::move(_astID));
+		return std::make_shared<DebugData const>(
+			std::move(_irLocation),
+			std::move(_originalLocation),
+			std::move(_astID)
+		);
 	}
 
-	langutil::SourceLocation location;
+	/// Location in the Yul code.
+	langutil::SourceLocation irLocation;
+	/// Location in the original source that the Yul code was produced from.
+	/// Optional. Only present if the Yul source contains location annotations.
+	langutil::SourceLocation originalLocation;
```

Comment:
What about `originLocation?

Reply:
Sounds fine to me. I'd be even fine with `origin` but I'm not sure if it would then be clear enough that we're dealing with location.